api/src/test/java/org/pmiops/workbench/actionaudit/targetproperties/WorkspaceTargetPropertyTest.kt

```diff
@@ -0,0 +1,133 @@
+package org.pmiops.workbench.actionaudit.targetproperties
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.pmiops.workbench.model.DataAccessLevel
+import org.pmiops.workbench.model.ResearchPurpose
+import org.pmiops.workbench.model.Workspace
+import org.springframework.test.context.junit4.SpringRunner
+
+@RunWith(SpringRunner::class)
+class WorkspaceTargetPropertyTest {
+
+    private var workspace1: Workspace? = null
+    private var workspace2: Workspace? = null
+    private var emptyWorkspace: Workspace? = null
+
+    @Before
+    fun setUp() {
+        val researchPurpose1 = ResearchPurpose()
+        researchPurpose1.intendedStudy = "stubbed toes"
+        researchPurpose1.additionalNotes = "I really like the cloud."
+
+        val researchPurpose2 = ResearchPurpose()
+        researchPurpose2.intendedStudy = "broken dreams"
+        researchPurpose2.additionalNotes = "I changed my mind"
+        researchPurpose2.anticipatedFindings = "a 4-leaf clover"
+
+        val now = System.currentTimeMillis()
+
+        workspace1 = Workspace()
```

Comment:
(Opt / style nit) Is there a reason not to use the fluent interface here, e.g.

Workspace()
  .name("Workspace 1")
  .id("fc-id-1")
  .namespace("aou-rw-local1-")
  ...

Reply:
Oh yeah, I forgot about that. It turns out to have a pretty nice pattern.

The way that works in Kotlin is to use `apply()`. This lets me write key/value pairs as if it were a Kotlin constructor.

```
        workspace1 = Workspace()
                .apply { name = "Workspace 1" }
                .apply { id = "fc-id-1" }
                .apply { namespace = "aou-rw-local1-c4be869a" }
```